### ISSUE TITLE
mixed 64/32bit gcc5 dependency

### ISSUE THREAD

**Issue by eMPee584** [damentz/liquorix-package#1]
Please change the linux-headers dependency to `Depends: gcc-5 | linux-compiler-gcc-5-x86` to not break mixed 64/32bit systems.

**Comment by damentz** [comment]
@eMPee584 , per the email you sent in, you want this change because you run a mixed system with a 64-bit kernel on 32-bit userland.

I would actually prefer not changing this since I would need to dynamically depend on the linux-compiler-gcc-5-x86 package that I actually didn't use to build the kernel.  In my opinion, that will be confusing.

Also, I don't believe the 64-bit kernel provides that much of an advantage over the 32-bit / 32-bit + pae kernels on 32-bit userland.  32-bit applications can only allocate up to 4gb of ram, so you can get the affects of the 64-bit kernel, but by using the PAE enabled 32-bit kernel instead.

And last, if performance is the matter, you can always re-build the package for yourself and adjust the architecture specifically for your CPU.  That should improve performance significantly over the generic 32-bit kernel.

**Comment by eMPee584** [comment]
Runtime dependencies are not meant to show what a package has been built with, but to enable running them. I don't understand the problem with this change, as it has zero impact for anyone else.
I have 24GiB of RAM and do not want to have the heavy PAE for this much memory. While I am planning to migrate my system to 64bit whenever I get to it, for now, I run 32bit userland. And the only way to run the 64bit liquorix kernel on my system is if you add the `| linux-compiler-gcc-6-x86` bit to the `Depends` field, or I create a mock gcc-6:amd64 package with equivs.
Of course I can build my own kernel, but what is the point of this package than?

> Distribution Kernel Drop-in Replacement: Proper distribution style configuration supporting broadest selection of hardware.


 Basically I just wanted to do a quick test of the MuQSS scheduler when I added the liquorix repo half a year ago. But meh.

**Comment by eMPee584** [comment]
The purpose of this package is exactly to prevent this situation. All other linux-headers packages in debian depend on the metapackage, instead of gcc-6.
```
Package: linux-compiler-gcc-6-x86 
Depends: gcc-6 
Description: Compiler for Linux on x86 (meta-package) 
 This package depends on gcc 6 of the appropriate architecture for Linux on 
 amd64, i386 and x32. 

```

**Comment by damentz** [comment]
So, I took a look at this issue again, and it appears that your solution wouldn't work even if the dependencies matched.

You cannot use a 32-bit compiler to build modules for a 64-bit kernel.  In fact, the way it works is the kernel header depends on the gcc version that was used to build the original kernel.  If the dependencies were to allow your system to install the 32-bit kernel, _all_ your kernel module builds through DKMS or third party build systems will fail.

Furthermore, if you need to run a 64-bit kernel with 32-bit user land, don't install the kernel headers if your hardware allows it.  Otherwise, maybe use a 32-bit kernel with PAE support?  That's why it's still offered as a download since it specifically caters to your use case.

**Comment by metux** [comment]
@damentz: running 64bit kernel w/ 32bit userland does indeed have valid use cases, eg. when host system is 32bit, but containers w/ 64bit.
